fhir-server-test/src/test/java/com/ibm/fhir/server/test/ReadTest.java

```diff
@@ -0,0 +1,155 @@
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient responsePatient = response.readEntity(Patient.class);
+        TestUtil.assertResourceEquals(patient, responsePatient);
+    }
+
+    @Test(groups = { "server-read" }, dependsOnMethods = {"testCreatePatient" })
+    public void testReadPatient_elements() throws Exception {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient/" + patientId)
+                .queryParam("_elements", "active")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient patient = response.readEntity(Patient.class);
+        assertNotNull(patient);
+        checkSubsetted(patient, Collections.singletonList("getActive"));
+    }
+
+    @Test(groups = { "server-read" }, dependsOnMethods = {"testCreatePatient" })
+    public void testReadPatient_summary() throws Exception {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient/" + patientId)
+                .queryParam("_summary", "true")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient patient = response.readEntity(Patient.class);
+        assertNotNull(patient);
+        checkSubsetted(patient, Arrays.asList("getActive", "getBirthDate", "getIdentifier", "getGender",
+            "getLink", "getManagingOrganization", "getName", "getTelecom"));
+    }
+
+    @Test(groups = { "server-read" }, dependsOnMethods = {"testCreatePatient" })
+    public void testReadPatient_elements_summary() throws Exception {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient/" + patientId)
+                .queryParam("_elements", "active")
+                .queryParam("_summary", "true")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient patient = response.readEntity(Patient.class);
+        assertNotNull(patient);
+        checkSubsetted(patient, Collections.singletonList("getActive"));
+    }
+
+    @Test(groups = { "server-read" }, dependsOnMethods = {"testCreatePatient" })
+    public void testReadPatient_nonGeneralSearchParameter() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient/" + patientId)
+                .queryParam("_lastUpdated", "ge2020-11-02")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+            "Read only supports general search parameters.");
```

Comment:
I wonder if we should just list them here since there's only 2.  Is "general search parameters" something defined in the spec?

Reply:
There's a section in the spec called General parameters, https://www.hl7.org/fhir/http.html#parameters, for ones that apply to any interaction. (_format, _pretty, _summary, _elements)  I could just mention _summary and _elements.  Just wasn't sure what would be the least confusing way to phrase it.

Reply:
ah, i do think we support `_format` and `_pretty` for all requests (including reads).  if so, i think your language is good.

Reply:
sorry, thought about this some more and I still think we should either:
A. list the supported parameters; or
B. tell the user which parameter they used which is invalid

Reply:
Updated to indicate the invalid parameter(s).